modules/sdk/src/main/scala/org/tessellation/sdk/cli/snapshot.scala

```diff
@@ -29,14 +29,16 @@ object snapshot {
     .env[Path]("CL_SNAPSHOT_INFO_PATH", help = "Path to store snapshot infos")
     .withDefault(Path("data/snapshot_info"))
 
-  val paths = (snapshotPath, incrementalPersistedSnapshotPath, incrementalTmpSnapshotPath, snapshotInfoPath)
-
-  val opts = paths.tupled.mapValidated {
-    case (snapshotPath, incrementalPersistedSnapshotPath, incrementalTmpSnapshotPath, snapshotInfoPath)
-        if Set(snapshotPath, incrementalPersistedSnapshotPath, incrementalTmpSnapshotPath, snapshotInfoPath).size.toLong === paths.size =>
-      (snapshotPath, incrementalPersistedSnapshotPath, incrementalTmpSnapshotPath, snapshotInfoPath).validNel[String]
-    case _ =>
-      "Paths for global snapshot and incremental snapshot (both persisted and tmp) must be different.".invalidNel[(Path, Path, Path, Path)]
+  val pathsOpts = (snapshotPath, incrementalPersistedSnapshotPath, incrementalTmpSnapshotPath, snapshotInfoPath)
+  val pathsErr = "Paths for file storages must differ.".invalidNel[(Path, Path, Path, Path)]
+
+  val opts = pathsOpts.tupled.mapValidated {
+    case (snapshotPath, incrementalPersistedSnapshotPath, incrementalTmpSnapshotPath, snapshotInfoPath) =>
+      val paths = List(snapshotPath, incrementalPersistedSnapshotPath, incrementalTmpSnapshotPath, snapshotInfoPath)
+      if (paths.size === paths.toSet.size)
+        (snapshotPath, incrementalPersistedSnapshotPath, incrementalTmpSnapshotPath, snapshotInfoPath).validNel[String]
```

Comment:
Does this refactor work? I'm trying to avoid the same tuple repeated 3 times
```suggestion
    case pathsTupled =>
       if (pathsTupled == pathsTupled.distinct)
         pathsTupled.validNel[String]
```

Reply:
I forgot the `.toList` part in my suggestion.

Reply:
there is no `.toList` on tuple. I've used the shapeless way right now. please re-review
![image](https://github.com/Constellation-Labs/tessellation/assets/4959931/18101001-811a-4872-a5b9-668323690e4d)

Reply:
Looks like my suggestion will only work for Scala 3. Disregard...